core/src/main/java/org/infinispan/factories/InterceptorChainFactory.java

```diff
@@ -12,28 +12,9 @@
 import org.infinispan.configuration.cache.InterceptorConfiguration;
 import org.infinispan.configuration.cache.StoreConfiguration;
```

Comment:
Yes, InterceptorChain is deprecated, but it's still available. I've added an explanation in the javadoc on why we can't have two separate factories.